data/data/install.openshift.io_installconfigs.yaml

```diff
@@ -434,6 +434,58 @@ spec:
                           - high_performance
                           type: string
                       type: object
+                    powervs:
+                      description: PowerVS is the configuration used when installing
+                        on IBM Power VS.
+                      properties:
+                        imageName:
+                          description: ImageName defines the boot image name for the
+                            instance.
+                          type: string
+                        keypairname:
+                          description: KeyPairName is the name of an SSH key pair
+                            stored in the Power VS Service Instance
+                          type: string
+                        memory:
+                          description: Memory defines the memory in GB for the instance.
+                          type: integer
+                        name:
+                          description: Name is the name of the instance
+                          type: string
+                        networkIDs:
+                          description: NetworkIDs defines the network IDs of the instance.
+                          items:
+                            type: string
+                          type: array
+                        procType:
+                          description: ProcType defines the processor sharing model
+                            for the instance.
+                          type: string
+                        processors:
+                          description: 'Processors defines the processing units for
+                            the instance. @TODO:'
+                        serviceinstance:
+                          description: ServiceInstance is Service Instance to install
+                            into.
+                          type: string
+                        sysType:
+                          description: SysType defines the system type for instance.
+                          type: string
+                        volumeIDs:
+                          description: VolumeIDs is the list of volumes attached to
+                            the instance.
+                          items:
+                            type: string
+                          type: array
+                      required:
+                      - imageName
```

Comment:
imageName shouldn't be required. this should come from the rhcos stream json once we have the pipeline setup and generating images

Reply:
hmm..are these the boot images and not the ovas? i think long term you want the bootimage creation handled by TF or by a separate component in the installer itself by consuming the ova image i would think. in any case because ovas are what are coming from the pipeline, this needs to be optional.

Reply:
I'm not clear on how to make this file show what we'd like to be required in the install-config.yaml. These seem to be pulled from the Config (type MachinePool) in the install config. Those items _are_ required (see https://github.com/openshift/terraform-provider-ibm/blob/master/website/docs/r/pi_instance.html.markdown). But they're not required in the install-config.yaml itself. 

To answer your questions though, -- yes, they should come from the rhcos stream json by default; and yes, they are the boot images; and yes, long-term the bootimage creation will be handled by TF.

Reply:
ah. i just wasn't looking at the file as a whole. there's a compute section, a machinepools section, and a platform section all for powervs. and i didn't understand how the +optional directives would be used for each of those things w/in the code. after looking more closely, iiuc, every time i listed something as +optional, it was referring to it being defined in the install config. so i'll go band and update those. 

this leads to the question though -- for everything that shows up in this yaml (install.openshift.io_installconfigs.yaml) -- we don't have to make it configurable via the install-config.yaml do we?

Reply:
so this is my view - things that can be configured optionally (for example through providerspec in the manifests) should not even be included in the install config. the user can modify them after doing a `openshift-install create manifests` and modifying the provider spec which will then be pulled in by the tfvars for master and for workers, it will be handled by the cluster api provider. so IMO those options shouldn't even be exposed here. Only things which are absolutely essential to change and makes sense to do so should be mentioned here like image name, region and even for that we need to have defaults. basically we are trying to make the users life easier and not complicate it by providing too many options even though they are optional.

Reply:
so tl;dr is don't even define the ones that are optional in `pkg/types/powervs/platform.go` . they can be changed through the provider spec defined in cluster-api-provider-powervs. only add things to the platform.go that absolutely must be exposed to the user through install-config